src/wildcard-service/init.lua

```diff
@@ -6,4 +11,67 @@ local mt = {
   __index = _M
 }
 
+function _M.new(options)
+  local opts = options or {}
+  local backend = cache_backend.new(opts.client or resty_backend)
+
+  local http_client = http_ng.new({
+    backend = backend,
+    options = { ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') or false } }
+  })
+
+  local api_host = opts.api_host or getenv('API_HOST') or 'system-provider:3000'
+  local access_token = opts.access_token or getenv('ACCESS_TOKEN')
+
+  return setmetatable({
+    options = opts,
+    http_client = http_client,
+    api_host = api_host,
+    access_token = access_token
+  }, mt)
+end
+
+function _M:get_apicast_servers()
+  local arg_host = _M.arg_host()
+  local query = ngx.encode_args({ host = arg_host, access_token = self.access_token })
+
+  if not arg_host then
+    ngx.log(ngx.WARN, 'missing host')
+    return ngx.exit(404)
+  end
+
+  for _, env in ipairs({'production', 'sandbox'}) do
+    local api_url = self.api_host .. '/admin/api/services/proxy/configs/' .. env .. '.json?' .. query
+    local response = self.http_client.get(api_url)
+
+    if response.status == 200 then
+      configs = cjson.decode(response.body)
+      if #configs.proxy_configs > 0 then
+        local servers = {}
+        for _, proxy_config in ipairs(configs.proxy_configs) do
+          local proxy = cjson.decode(proxy_config.content).proxy
+          if env == 'production' then
+            servers[#servers + 1] = proxy.endpoint
+          else
+            servers[#servers + 1] = proxy.sandbox_endpoint
+          end
+        end
+        return servers
+      end
+    else
+      ngx.log(ngx.ERR, 'failed to load apicast config')
```

Comment:
might be good to print the response status and even the body, because the request should always return success right?